sbt-bridge/src/dotty/tools/xsbt/PositionBridge.java

```diff
@@ -116,4 +116,9 @@ public String lineContent() {
       result.append(lineContent.charAt(i) == '\t' ? '\t' : ' ');
     return Optional.of(result.toString());
   }
+
+  @Override
+  public String toString() {
+      return pos;
```

Comment:
This won't typecheck since pos isn't a String, you need pos.toString at least, but also I don't know if pos.toString actually return the line number or if you should get that manually to match the output of scala 2 here.

Reply:
This file also defines `noPosition` where you also probably want to set the toString to the empty String to match what Scala 2 does.

Reply:
Excuse me for this stupid oversight.

I added empty String impl to noPosition, thx for the pointer.

As for the PositionBridge.toString(), I would lean on the implementations of the toString of SourcePosition subtypes, instead of custom implementing it with the risk of being incomplete. The more information about where the compile problem is, the better.